src/ProgressOnderwijsUtils/Data/SqlBatch.cs

```diff
@@ -122,6 +123,29 @@ public ScalarSqlCommand<T> WithTimeout(CommandTimeout timeout)
     }
 }
 
+public readonly record struct MaybeSqlCommand<TOut, TCommand>(TCommand underlying) : ITypedSqlCommand<Maybe<TOut, Exception>, MaybeSqlCommand<TOut, TCommand>>
+    where TCommand : ITypedSqlCommand<TOut, TCommand>
+{
+    public MaybeSqlCommand<TOut, TCommand> WithTimeout(CommandTimeout timeout)
+        => new(underlying.WithTimeout(timeout));
+
+    [MustUseReturnValue]
+    public Maybe<TOut, Exception> Execute(SqlConnection conn)
+    {
+        try {
+            return Maybe.Ok(underlying.Execute(conn));
+        } catch (Exception e) {
+            return Maybe.Error(e);
+        }
```

Comment:
Waarom geen Maybe.Try().Catch() gebruikt hier?

Reply:
Het levert vrij weinig op, maar sure, zou kunnen. Als het al een debugger hulpje is, dan is de "echte" catch zo iets duidelijker.

Mij om 't even.

Reply:
Mij ook ja

Reply:
dan hou ik hem zo.